### DISCUSSION TITLE
Disable OidcClientFilter at runtime for simpler application testing

### DISCUSSION THREAD

**Discussion by lmartella1** [quarkusio/quarkus#37283] [Q&A]
Hello,
in our Quarkus application we rely on the _OidcClientFilter_ to instrument Quarkus in acquiring a token before calling a given REST Client.
In certain cases, such as when we run quarkus tests or we deploy the application in a local environment, we believe it could be useful to disable the OidcClientFilter in a way that the token is not acquired and the client call is made to the remote (in many cases mocked) server without any bearer token. In all these cases, is more relevant to test the application business logic than the actual token acquisition and security aspects.

Looking at the current available configuration for the oidc-client, I don't see any runtime-overridable configuration that can serve this scope and instrument Quarkus to just ignore the _OidcClientFilter_ annotation either globally, for all the oidc clients, or specifically for a given named oidc client.

What is your vision about this use case? Is it a feature that could be implemented in a future version of the framework? 

For reference, we have already played with :
- `quarkus.oidc-client.enabled=false` which works globally in disabling the token acquisition for all the annotated _OidcClientFilter_  REST Clients but which is fixed at build time and cannot be overridden at runtime. 
- `quarkus.oidc-client."id".client-enabled=false` which is disabling a given oidc-client configuration in a way that, if the client is used then the following runtime exception is raised: `io.quarkus.oidc.client.runtime.DisabledOidcClientException`. 

Thanks in advance,

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment]
Hi @lmartella1

I think the fact that `OidcClientFilter` throws an exception when the client is disabled at runtime is a bug, it should log the debug message and just return return

**Comment by sberyozkin** [comment]
@lmartella1 
https://github.com/quarkusio/quarkus/blob/main/extensions/oidc-client-filter/runtime/src/main/java/io/quarkus/oidc/client/filter/runtime/AbstractOidcClientRequestFilter.java#L29 should just return
and
https://github.com/quarkusio/quarkus/blob/main/extensions/oidc-client-reactive-filter/runtime/src/main/java/io/quarkus/oidc/client/reactive/filter/runtime/AbstractOidcClientRequestReactiveFilter.java#L42 should return too after `requestContext.resume()`

If you'd like please open a simple PR, I can have look but it will be in a few days or so

**Comment by sberyozkin** [comment]
@lmartella1 It has been fixed

**Reply by lmartella1** [reply]
Great! Thank you very much!

**Comment by sahil24sharma** [comment]
@sberyozkin I don't see it fixed. What dependency I need to add to see the changes. I added   `implementation 'io.quarkus:quarkus-oidc-client-reactive-filter:3.7.1'.

**Comment by sberyozkin** [comment]
@sahil24sharma It was fixed with this PR, https://github.com/quarkusio/quarkus/pull/37361/files, `DisabledOidcClientException` is now not blocking the filter flow. What is happening in your case ?

**Reply by sahil24sharma** [reply]
@sberyozkin in my `_AbstractOidcClientRequestReactiveFilter_` i can see which is aborting my test.

```
 if (t instanceof DisabledOidcClientException) {
                    LOG.debug("Client is disabled");
                    requestContext.abortWith(Response.status(Response.Status.INTERNAL_SERVER_ERROR).build());
                }
```

**Reply by sberyozkin** [reply]
@sahil24sharma Looks like you are picking up the old code, on 3.7.1 it is not aborting the request: https://github.com/quarkusio/quarkus/blob/3.7.1/extensions/oidc-client-reactive-filter/runtime/src/main/java/io/quarkus/oidc/client/reactive/filter/runtime/AbstractOidcClientRequestReactiveFilter.java#L41
